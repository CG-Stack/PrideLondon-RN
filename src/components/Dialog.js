// @flow
import React from "react";
import type { Node } from "react";
import {
  Modal,
  StyleSheet,
  TouchableWithoutFeedback,
  View
} from "react-native";
import Button from "./ButtonPrimary";
import {
  dialogBackdropColor,
  dialogBgColor,
  dialogHeaderDividerColor
} from "../constants/colors";

type Props = {
  applyButtonText: string,
  applyButtonLabel: string,
  children: Node,
  headerLeft?: Node,
  headerRight?: Node,
  onApply: Function,
  onCancel?: Function,
  title: Node,
  visible: boolean
};

const Dialog = ({
  applyButtonText,
  applyButtonLabel,
  children,
  headerLeft,
  headerRight,
  onApply,
  onCancel,
  title,
  visible
}: Props) => (
  <Modal
    animationType="fade"
    onRequestClose={onCancel}
    transparent
    visible={visible}
  >
    <TouchableWithoutFeedback
      style={styles.backdrop}
      onPress={onCancel}
      accessible={false}
    >
      <View style={styles.container}>
        <TouchableWithoutFeedback accessible={false}>
          <View style={styles.content}>
            <View style={styles.header}>
              <View style={[styles.headerSide, styles.headerSideLeft]}>
                {headerLeft}
              </View>
              <View style={[styles.headerSide, styles.headerSideMiddle]}>
                {title}
              </View>
              <View style={[styles.headerSide, styles.headerSideRight]}>
                {headerRight}
              </View>
            </View>
            {children}
          </View>
        </TouchableWithoutFeedback>
        <View style={styles.applyButtonContainer}>
          <Button onPress={onApply} accessibilityLabel={applyButtonLabel}>
            {applyButtonText}
          </Button>
        </View>
      </View>
    </TouchableWithoutFeedback>
  </Modal>
);

Dialog.defaultProps = {
  headerLeft: undefined,
  headerRight: undefined,
  onCancel: () => {}
};

const styles = StyleSheet.create({
  backdrop: {
    flex: 1
  },
  container: {
    flex: 1,
    justifyContent: "center",
    padding: 16,
    backgroundColor: dialogBackdropColor
  },
  content: {
    backgroundColor: dialogBgColor,
    borderRadius: 4,
    paddingBottom: 8
  },
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    height: 40,
    paddingHorizontal: 16,
    borderBottomWidth: 1,
    borderBottomColor: dialogHeaderDividerColor
  },
  headerSide: {
    width: 0,
    flexGrow: 1
  },
  headerSideLeft: {
    alignItems: "flex-start"
  },
  headerSideMiddle: {
    alignItems: "center"
  },
  headerSideRight: {
    alignItems: "flex-end"
  },
  applyButtonContainer: {
    marginTop: 8
  }
});

export default Dialog;
