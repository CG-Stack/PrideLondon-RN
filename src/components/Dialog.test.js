// @flow
import React from "react";
import { shallow } from "enzyme";
import Dialog from "./Dialog";

it("renders correctly", () => {
  const output = shallow(
    <Dialog
      applyButtonText="Bye"
      onApply={() => {}}
      title="Hello World"
      visible
    >
      Lorem Ipsum is placeholder text commonly used...
    </Dialog>
  );
  expect(output).toMatchSnapshot();
});
