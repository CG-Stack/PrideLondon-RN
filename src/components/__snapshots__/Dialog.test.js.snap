// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "padding": 16,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "paddingBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#F2F2F2",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexGrow": 1,
                "width": 0,
              },
              Object {
                "alignItems": "flex-start",
              },
            ]
          }
        />
        <Text
          markdown={false}
          style={
            Object {
              "color": "#272727",
            }
          }
          type="h3"
        >
          Hello World
        </Text>
        <View
          style={
            Array [
              Object {
                "flexGrow": 1,
                "width": 0,
              },
              Object {
                "alignItems": "flex-end",
              },
            ]
          }
        />
      </View>
      Lorem Ipsum is placeholder text commonly used...
    </View>
    <TouchableOpacity
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(51, 51, 51)",
          "borderRadius": 4,
          "height": 48,
          "justifyContent": "center",
          "marginTop": 8,
        }
      }
    >
      <Text
        markdown={false}
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
        type="text"
      >
        Bye
      </Text>
    </TouchableOpacity>
  </View>
</Component>
`;
